core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -452,16 +452,23 @@ public void removeBlocks(List<Long> blockIds, boolean delete) throws Unavailable
   @Override
   public void validateBlocks(Function<Long, Boolean> validator, boolean repair)
       throws UnavailableException {
-    List<Long> invalidBlocks = new ArrayList<>();
-    for (long blockId : mBlocks.keySet()) {
-      if (!validator.apply(blockId)) {
-        invalidBlocks.add(blockId);
+    List<Long> invalidBlocks = mBlocks.keySet()
+        .stream()
+        .filter((blockId) -> !validator.apply(blockId))
+        .collect(Collectors.toList());
+    if (!invalidBlocks.isEmpty()) {
+      long maxSize = 100;
+      List<Long> loggedBlocks = invalidBlocks.stream().limit(maxSize).collect(Collectors.toList());
+      LOG.warn("Found {} blocks (including {}) have no corresponding file metadata.",
+          invalidBlocks.size(), loggedBlocks);
```

Comment:
This message format will be hard to read since the 100 blocks are logged in the middle of the message. Could we log the blocks at the end?

Reply:
fixed